fix bug - unfinished reportback influencing new reportback flow, add tests

tongxiang opened the pull request:
#### What's this PR do?
Previously, if a reportback DB document was created but the reportback was never submitted/finished by the user, it was never destroyed and thus lingered in the database. 

If a user didn't finish a reportback flow, but then wanted to reportback into another campaign, our app would sometimes find their previous unfinished reportback documents and update those documents, messing up the new reportback flow. 

This PR adds a new, campaign-specific param to our DB calls to eliminate this interference. 

#### Where should the reviewer start?
`reportback/index.js` contains the modified code. 

#### How should this be manually tested?
By beginning a reportback flow (1/3 Postman collection [here](https://www.getpostman.com/collections/71bd60b99e46cc8ce514
)), stopping at any stage before you submit the reportback, and then creating a new reportback flow. (Comeback Clothes Postman collection [here](https://www.getpostman.com/collections/8ce30f8111deea794ba9)). 

Created #443 to add test coverage. 

#### What are the relevant tickets?
Closes #442, #443.

jonuy commented on app/lib/reportback/test/reportback.test.js:
```diff
@@ -6,12 +6,15 @@ var assert = require('assert')
-  var TEST_CAMPAIGN_CONFIG = app.getConfig(app.ConfigName.REPORTBACK, 'test', 'endpoint');
+  var TEST_PHONE = '15555555555'
+    , TEST_CAMPAIGN_CONFIG = app.getConfig(app.ConfigName.REPORTBACK, 'test', 'endpoint')
+    , testDoc // exists in the global test scope; variable re-assigned each time createTestDoc() is run. 
```
Instead of pulling this var up to the global scope, I think I'd prefer we continue using the `testDoc` vars in their previously more local scope and just add the campaign property. **Or** to pass in `testData` to both parameters - ex: `reportback.receivePhoto(testData, testData);`

Having this particular mutable variable in the global scope can be confusing without context of the entire setup of tests in this file. Like, I can see `before(createTestDoc);` in a test, but not necessarily know intuitively that the `testDoc` var has now just been changed.

tongxiang commented:
Changed the global `testDoc` document to be local to each test, added tests for the situation where a user submits one reportback, doesn't finish it, and then begins another. @jonuy

jonuy commented on app/lib/reportback/test/reportback.test.js:
```diff
@@ -337,12 +371,143 @@ function test() {
+      reportback.handleUserResponse(testDoc2, quantityData);
+    })
+
+    after(function() {
```
Seems like this could just be `after(removeTestDocs);`

jonuy commented:
:+1:
